vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -668,6 +664,49 @@ public MongoClient dropIndex(String collection, String indexName, Handler<AsyncR
     return this;
   }
 
+  @Override
+  public MongoClient watch(String collection, JsonArray pipeline, WatchOptions options, Handler<AsyncResult<MongoClientWatchIterable>> resultHandler) {
+    requireNonNull(collection);
+    requireNonNull(pipeline);
+
+    if (options == null) {
+      options = new WatchOptions();
+    } else {
+      if (options.getFullDocument() != null
+        && !options.getFullDocument().equals(WatchOptions.FULL_DOCUMENT_UPDATE_LOOKUP)) {
+        resultHandler.handle(Future.failedFuture(new IllegalArgumentException("fullDocument must be null or \"updatedLookup\"")));
+        return this;
+      }
+    }
+
+    try {
+      List<Bson> pipelineBson = new ArrayList<>();
+      for (int i = 0; i < pipeline.size(); i++) {
```

Comment:
can we use an enhanced for loop here?

Reply:
This should actually probably use the `JsonObjectBsonAdapter`